### ISSUE TITLE
@nuxt/typescript-build cannot check types for literal strings

### ISSUE THREAD

**Issue by szulcus** [nuxt/typescript#555]
**Describe the bug**
I am getting an unnecessary error: `No index signature with a parameter of type 'string' was found on type 'Data & ...'.`

**To Reproduce**
1. Write in component:
```
interface Data {
	userDetailsChanged: boolean;
	organizationDetailsChanged: boolean;
	resourceDetailsChanged: boolean;
}
---
methods: {
	changeDetailsInfo(id: 'user' | 'organization' | 'resource', changed: boolean) {
		this[`${id}DetailsChanged`] = changed;
	}
},
```
2. See error

**Expected behavior**
I expect ESLint compatibility.

**Screenshots**
It works:
![image](https://user-images.githubusercontent.com/48328739/180993027-1105378b-39d5-4ebb-9395-e54bafcf90d3.png)

It doesn't work:
![image](https://user-images.githubusercontent.com/48328739/180992985-8e66559a-da53-434e-9dc0-43ab5b897d55.png)

**Comment by danielroe** [comment]
I'm afraid this is TypeScript behaviour, and is not related to Nuxt's integration.

Try:

```ts
type Data = {
	userDetailsChanged: boolean;
	organizationDetailsChanged: boolean;
	resourceDetailsChanged: boolean;
	[key: string]: boolean;
}
```

**Comment by szulcus** [comment]
> I'm afraid this is TypeScript behaviour, and is not related to Nuxt's integration.
> 
> Try:
> 
> ```ts
> type Data = {
> 	userDetailsChanged: boolean;
> 	organizationDetailsChanged: boolean;
> 	resourceDetailsChanged: boolean;
> 	[key: string]: boolean;
> }
> ```

I understand, then I'll stick with the current workaround. Thank you for your response.
